### PR TITLE
Make FillAntibot faster

### PR DESCRIPTION
**Pull request by def-**
## Checklist

- [ ] Tested the change ingame
- [ ] Provided screenshots if it is a visual change
- [ ] Tested in combination with possibly related configuration options
- [ ] Written a unit test (especially base/) or added coverage to integration test
- [ ] Considered possible null pointers and out of bounds array indexing
- [ ] Changed no physics that affect existing maps
- [ ] Tested the change with [ASan+UBSan or valgrind's memcheck](https://github.com/ddnet/ddnet/#using-addresssanitizer--undefinedbehavioursanitizer-or-valgrinds-memcheck) (optional)

### CODE REVIEW

File: src/engine/server/server.cpp [hunk 1]
```diff
@@ -2285,7 +2285,9 @@ void CServer::FillAntibot(CAntibotRoundData *pData)
 	for(int i = 0; i < MAX_CLIENTS; i++)
 	{
 		CAntibotPlayerData *pPlayer = &pData->m_aPlayers[i];
-		str_copy(pPlayer->m_aAddress, m_NetServer.ClientAddrString(i));
+		// No need for expensive str_copy since we generated this string ourselves
+		// and it is known to be valid UTF-8
+		mem_copy(pPlayer->m_aAddress, m_NetServer.ClientAddrString(i), sizeof(pPlayer->m_aAddress));
```
Comment (heinrich5991) [inline src/engine/server/server.cpp:2290]: Why is this not a buffer overread of `m_NetServer.ClientAddrString(i)`?

Comment (heinrich5991) [reply src/engine/server/server.cpp:2290]: Note that the `str_utf8_fix_truncation` is only used to fix truncation of an incomplete UTF-8 code sequence. `str_copy` already assumes that its inputs are UTF-8.

The correct reason why this truncated string is still valid UTF-8 is that it's known to be ASCII-only.

Comment (def-) [reply src/engine/server/server.cpp]: True, fixed.

**Comment by heinrich5991** [comment]
(Otherwise, this looks very brittle and unlikely to get noticed if this suddenly became a buffer overread/overflow.)

**Comment by def-** [comment]
Done, I'm not good at C++ anymore though, not sure if this it clean.

**Review by heinrich5991 — APPROVED** [review]
Thanks!

**Pull request merged**